### PR TITLE
RoR group project: Recipe App

### LINKED ISSUE #8: [1pt] Set up RSpec and Capybara testing libraries
- Add Rspec gem
- Add Capybara gem

### LINKED ISSUE #9: [1pt] Solve N+1 queries problem
- Make sure there are no N+1 queries happening to all different views.

### PR DESCRIPTION
**Pull request by MarcoOquendoC**
# [RoR group project: Recipe App](https://github.com/microverseinc/curriculum-rails/blob/main/recipe-app/buisness_requirements.md)

 ## Checklist
- [x] I have read this repository's contribution guidelines.
- [x] My pull request has a descriptive title (not a vague title like Update index.MD).
- [x] My pull request targets the main branch of this [project](https://github.com/didierganthier/recipe-app).
- [x] I added relevant documentation for all the files I changed.

### General requirements
- [x] No linter errors.
- [x] Use correct Gitflow.
- [x] Documentation in a professional way.
### Ruby requirements
- [x] Follow best practices for Ruby.
### Project requirements
- [x] Follow the layout of the wireframes [provided](https://github.com/microverseinc/curriculum-rails/blob/main/recipe-app/buisness_requirements.md) by Microverse.
- [x] Use Devise to login and register.
- [x] Food and Recipes lists functionalities following the [description](https://github.com/microverseinc/curriculum-rails/blob/main/recipe-app/buisness_requirements.md) provided. 
- [x] Public recipe list.
- [x] Recipe details

## Authors
- @didierganthier 
- @MarcoOquendoC

### CODE REVIEW

File: app/controllers/foods_controller.rb [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+        test.quantity = test.quantity - recipe_food.quantity
+      end
+    end
+    @foods = @foods.select { |f| f.quantity.negative? }
+    @foods.each { |f| f.quantity *= -1 }
+    @total = 0
+    @foods.each do |food|
+      @total += (food.price * food.quantity)
+    end
+  end
```
Comment (gemmen29) [inline app/controllers/foods_controller.rb]: - [x] Kindly fix the `n+1` problem on the `general shopping list` page. You already use bullet to solve the problem but you didn't enable it. When I enable it I get multiple problems on this page.
image


### Hint: To enable it you can run the command
```
bundle exec rails g bullet:install
```
for more information, you can visit the gem documentation from [here](https://github.com/flyerhzm/bullet).

**Review by gemmen29 — CHANGES_REQUESTED** [review]
Hi Team, 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!

## Highlights 💯 
- No linters errors ✔️ 
- Descriptive pull request ✔️ 
- Descriptive commit messages ✔️ 

## Required Changes ♻️
_Check the comments under the review._



## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: app/controllers/foods_controller.rb (hunk 1 above)
Comment (MarcoOquendoC) [reply app/controllers/foods_controller.rb]: # Hi @gemmen29,
It is done, thank you.

**Review by prolajumokeoni — APPROVED** [review]
Hi @Team

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 
## Highlights
-  N+1 queries fixed 👏🏿 
- Good App design 💪🏿 

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

You can also consider:
- N/A

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification @prolajumokeoni.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Comment by didierganthier** [comment]
Thanks @prolajumokeoni you're the best ☺️

**Pull request merged**